BlockchainSdk/WalletManagers/Aptos/AptosTransactionBuilder.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  AptosTransactionBuilder.swift
+//  BlockchainSdk
+//
+//  Created by skibinalexander on 25.01.2024.
+//  Copyright © 2024 Tangem AG. All rights reserved.
+//
+
+import Foundation
+import WalletCore
+import TangemSdk
+import CryptoKit
+
+final class AptosTransactionBuilder {
+    private let publicKey: Data
+    private let decimalValue: Decimal
+    private let chainId: AptosChainId
+    
+    private var coinType: CoinType { .algorand }
+    
+    // MARK: - Init
+    
+    init(
+        publicKey: Data,
+        decimalValue: Decimal,
+        chainId: AptosChainId
+    ) {
+        self.publicKey = publicKey
+        self.decimalValue = decimalValue
+        self.chainId = chainId
+    }
+    
+    // MARK: - Implementation
+
+    func buildForSign(transaction: Transaction) throws -> Data {
+        let input = try buildInput(transaction: transaction)
+        let txInputData = try input.serializedData()
+
+        guard !txInputData.isEmpty else {
+            throw WalletError.failedToBuildTx
+        }
+
+        let preImageHashes = TransactionCompiler.preImageHashes(coinType: coinType, txInputData: txInputData)
+        let preSigningOutput = try TxCompilerPreSigningOutput(serializedData: preImageHashes)
+
+        guard preSigningOutput.error.rawValue == 0, !preSigningOutput.dataHash.isEmpty else {
```

Comment:
Вроде вот так можно, наверно не очень на `rawValue` завязываться 
```suggestion
        guard preSigningOutput.error == .ok, !preSigningOutput.dataHash.isEmpty else {
```

Reply:
> preSigningOutput.error

Cannot convert value of type 'TW_Common_Proto_SigningError' to expected argument type 'Int'